app/models/user.rb

```diff
@@ -30,6 +30,10 @@ def membership_payment_notes
     payment_notes(Payment::PAYMENT_TYPE_MEMBER)
   end
 
+  def membership_current?
```

Comment:
This makes me realize that we need to be sure to review all of the `...member...?` type methods so that we're certain that they are all synchronized.  For instance:  what is the relationship to this method (which looks at the expiration date) and the `member?` method -- which is based on the User model attribute.  Are they all going to give us consistent answers?
(nothing is wrong with this method.  Just making me think more about them all)

Reply:
That's a good question, and I wish we could just use _one_ attribute to determine membership.  Here, however, I can't - as payment status is changed independent of `member` status, but `member` status is set to false if payment status is expired.

Anyway, if `membership_current?` is `false`, we set`member` to `false` upon login.

So, no, we won't get consistent answers if we have an expired member fee payment, but the user has to logged in since expiration.

Longer term, we could run a job at midnight that set `member` based on payment expiration date.